orc8r/gateway/python/magma/magmad/upgrade/docker_upgrader.py

```diff
@@ -211,9 +212,10 @@ async def download_update(
         MAGMA_GITHUB_PATH, target_version,
     )
     await run_command(git_checkout_cmd, shell=True, check=True)
-    docker_login_cmd = "docker login -u $DOCKER_USERNAME -p $DOCKER_PASSWORD " \
-                       "$DOCKER_REGISTRY"
-    await run_command(docker_login_cmd, shell=True, check=True)
+    if os.getenv("DOCKER_USERNAME"):
```

Comment:
Where's this coming from / what does it indicate?

Reply:
I cannot run a docker login with docker repo with an anonymous user.
So this line of code is to permit the support of no credentials.